prov/verbs/src/fi_verbs.c

```diff
@@ -3177,12 +3221,26 @@ fi_ibv_domain(struct fid_fabric *fabric, struct fi_info *info,
 	   struct fid_domain **domain, void *context)
 {
 	struct fi_ibv_domain *_domain;
+	struct fi_info *fi;
 	int ret;
 
+	fi = fi_ibv_find_info(NULL, info->domain_attr->name);
+	if (!fi)
+		return -FI_EINVAL;
+
+	ret = fi_ibv_check_domain_attr(info->domain_attr, fi);
+	if (ret)
+		return ret;
+
 	_domain = calloc(1, sizeof *_domain);
 	if (!_domain)
 		return -FI_ENOMEM;
 
+	if (!(_domain->info = fi_dupinfo(fi))) {
+		ret = -FI_ENOMEM;
+		goto err;
+	}
```

Comment:
What is the reason to make a copy of fi_info here?  This shouldn't be needed.  The user passes in fi_info into any relevant call.  If needed, the domain can store a pointer to the fi_info in the list, but even that shouldn't be needed.